index.js

```diff
@@ -1,55 +1,82 @@
 const core = require('@actions/core');
 const { exec } = require('child_process');
-const https = require('https');
+const fetch = require('node-fetch');
 const fs = require('fs');
+const https = require('https');
 const path = require('path');
+const unzipper = require('unzipper');
 
-const token = core.getInput('token');
-
-const versionFilePath = path.join(__dirname, 'version.json');
-let version;
-
-if (core.getInput('version')) {
-  version = core.getInput('version');
-  console.log(`Using version from input: ${version}`);
-} else {
+async function main() {
   try {
-    const versionJson = fs.readFileSync(versionFilePath);
-    const defaultVersion = JSON.parse(versionJson).latest;
-    version = defaultVersion;
-    console.log(`Using default version from ${versionFilePath}: ${version}`);
-  } catch (err) {
-    console.error(`Error reading version file ${versionFilePath}: ${err.message}`);
-    process.exit(1);
-  }
-}
+    const token = core.getInput('token');
 
-const url = `https://github.com/typst/typst/releases/download/${version}/typst-x86_64-pc-windows-msvc.zip`;
-const downloadPath = `${process.env.RUNNER_TEMP}/typst.zip`;
+    const versionFilePath = path.join(__dirname, 'typst_version.json');
+    let version;
 
-await new Promise((resolve, reject) => {
-  const file = fs.createWriteStream(downloadPath);
-  https.get(url, { headers: { Authorization: `token ${token}` } }, (response) => {
-    response.pipe(file);
-    file.on('finish', () => {
-      file.close();
-      resolve();
-    });
-  }).on('error', (err) => {
-    fs.unlink(downloadPath, () => {});
-    reject(err);
-  });
-});
-
-await new Promise((resolve, reject) => {
-  exec(`7z x ${downloadPath} -o${process.env.RUNNER_TEMP}/typst`, (err) => {
-    if (err) {
-      reject(err);
+    if (core.getInput('version')) {
+      version = core.getInput('version');
+      console.log(`Using version from input: ${version}`);
     } else {
-      resolve();
+      try {
+        const versionJson = fs.readFileSync(versionFilePath, 'utf8');
+        const defaultVersion = JSON.parse(versionJson).latest;
+        version = defaultVersion;
+        console.log(`Using default version from ${versionFilePath}: ${version}`);
+      } catch (err) {
+        console.error(`Error reading version file ${versionFilePath}: ${err.message}`);
+        process.exit(1);
+      }
     }
-  });
-});
 
-const exePath = `${process.env.RUNNER_TEMP}/typst/typst.exe`;
-core.setOutput('path', exePath);
+    const url = `https://github.com/typst/typst/releases/download/${version}/typst-x86_64-pc-windows-msvc.zip`;
```

Comment:
It seems that the download path should be set here.
```suggestion
    const url = `https://github.com/typst/typst/releases/download/${version}/typst-x86_64-pc-windows-msvc.zip`;
    const downloadPath = `typst.zip`;
```